src/material/datepicker/calendar.ts

```diff
@@ -305,7 +310,7 @@ export class MatCalendar<D> implements AfterContentInit, AfterViewChecked, OnDes
               @Optional() private _dateAdapter: DateAdapter<D>,
               @Optional() @Inject(MAT_DATE_FORMATS) private _dateFormats: MatDateFormats,
               private _changeDetectorRef: ChangeDetectorRef,
-              private _model: MatDateSelectionModel<D | null, D>) {
+              private _model: MatDateSelectionModel<D | DateRange<D>, D>) {
```

Comment:
should this be `DateRange<D> | D | null` (since single selection model is `D | null`)